src/Collection/Collection.service.ts

```diff
@@ -1,21 +1,31 @@
 import { collectionAPI } from '../ethereum/api/collection'
 import { isPublished } from '../utils/eth'
-import { isManager as isTPWManger } from '../ethereum/api/tpw'
+import { thirdPartyAPI } from '../ethereum/api/thirdParty'
 import { FactoryCollection } from '../ethereum/FactoryCollection'
 import { Ownable } from '../Ownable'
-import { toDBCollection } from './utils'
+import { Item } from '../Item/Item.model'
+import { isManager } from '../ethereum/api/tpw'
+import { getThirdPartyCollectionURN, toDBCollection } from './utils'
 import { CollectionAttributes, FullCollection } from './Collection.types'
 import { Collection } from './Collection.model'
 
+enum CollectionType {
+  THIRD_PARTY,
+  DCL,
+}
 export class CollectionLockedException extends Error {
-  constructor(public id: string) {
-    super("The collection is locked. It can't be saved.")
+  constructor(public id: string, action: string) {
```

Comment:
I'm not entirely sold on the idea of having a generic string to serve as an error message. I think is a lot to put on the caller to know which action is being performed and to know exactly where the verb/word is going to end up in the error message.
What do you think of:
- Using an enum for each supported action
- Adding an error for each supported action which internally uses the same concept (just not exposed to the public API)

In conjunction to all of this, maybe we can move these errors to their own file?

Reply:
It seems to be a good idea! 
I'll do it in another PR so we don't have to do a big conflict resolution do to the multiple merges.